test/Npgsql.Tests/Support/PgPostmasterMock.cs

```diff
@@ -55,20 +64,17 @@ internal PgPostmasterMock(string? connectionString = null)
 
         void AcceptClients()
         {
+            _acceptingClients = true;
             _acceptClientsTask = DoAcceptClients();
 
             async Task DoAcceptClients()
             {
                 while (true)
                 {
                     var serverOrCancellationRequest = await Accept();
+                    await _pendingRequestsWriter.WriteAsync(serverOrCancellationRequest);
```

Comment:
I think, the test failed because startup and row description were written in parallel. If so, moving it below should fix this. Although it shouldn't happen, as until we've written all the messages from startup, the connection shouldn't be open.

Reply:
I think you're right. I think we have a race condition between the following:

* The WriteAsync unblocks WaitForServerConnection on the client. The test immediately gets the mock and writes some messages into the buffer.
* In parallel, the postmaster continues into Startup, which also reads and writes on the same server.

I'm still not sure why the test complained about getting AuthenticationRequest while expecting ParseCompleteMessage (i.e. how it even got to the point where it was expecting ParseCompleteMessage), but the bug is there - will fix.